src/watchdog.rs

```diff
@@ -0,0 +1,28 @@
+use registers::WDOG;
+use embedded_hal::watchdog;
+
+pub struct Watchdog {
+    wdog: WDOG,
+}
+
+pub trait WatchdogExt {
+    fn constrain(self) -> Watchdog;
+}
+
+impl WatchdogExt for WDOG {
+    fn constrain(self) -> Watchdog {
```

Comment:
I'm not sure what's the (future) intended semantics for this, a docstring would help. Also, I think you can skip the trait and just implement this method directly on the struct.

Reply:
I understand the sentiment, but I think `constrain` is a good convention here, it's documented [here](https://rust-embedded.github.io/book/start/registers.html#using-a-hal-crate), and while it's not a mandatory/standard API, as a convention it helps with the mental model that a PAC peripherals can be `constrain`ed to expose access to the higher HAL API.

Reply:
:+1:
(I didn't know about that docpage, I just learned something!)